patches/0002-macOS-Fix-install-name.patch

```diff
@@ -31,17 +38,113 @@ index 7b15709..7fa120c 100644
  
  ## Compiler switch to embed a runtime search path
 diff --git a/source/config/pkgdataMakefile.in b/source/config/pkgdataMakefile.in
-index f6fe198..68df3d2 100644
+index f6fe198..7934a8b 100644
 --- a/source/config/pkgdataMakefile.in
 +++ b/source/config/pkgdataMakefile.in
-@@ -16,6 +16,7 @@ include $(top_builddir)/icudefs.mk
+@@ -15,7 +15,9 @@ top_builddir = ..
+ include $(top_builddir)/icudefs.mk
+ 
+ OUTPUTFILE=pkgdata.inc
++LIB_EXT_ORDER:=$(FINAL_SO_TARGET)
+ MIDDLE_SO_TARGET=
```

Comment:
For reference, this makefile is "editing" the `LD_SONAME`, which is defined as:

```
LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name @rpath/$(notdir $(FINAL_SO_TARGET))
```

The existing code sets `MIDDLE_SO_TARGET` to nothing in order to erase that part of the `LIB_SONAME` variable for use by the `pkgdata` program, which then puts in its own idea of what the install name should be.

Since we're using `FINAL_SO_TARGET` as a substitution in `LD_SONAME` for macOS, so that the correct name is set into the dylibs, I had to change the code to erase that, too.

That broke the value of `LIB_EXT_ORDER` that was written to the file a few lines down. I used the `:=` to expand `FINAL_SO_TARGET` _immediately_, introducing the `LIB_EXT_ORDER` variable to bridge the value around the erasure of `FINAL_SO_TARGET`.

The whole "make a complicated variable then substitute part of it with nothing" was already kind of strange, as is having a makefile write a config file to be read by a tool written in C++ that then shells out to run the C++ compiler.